### PR TITLE
Fix towns visibility at z10-11

### LINKED ISSUE #1995: Better rendering priorities for towns
Here is a screenshot of the app with the map of the area to the south of Vienna, Austria. The largest towns/cities here are (north-to-south) Mödling, Baden, and Wiener Neustadt. However, *none* of the labels shown on the map names those. While I more or less know this area and can navigate there regardless, this would not be true for somewhere unknown.

In short: labels are apparently not ordered by priority. The app should show the most important labels when it doesn't have much space (low zoom level), not some random ones.

Mödling: https://www.openstreetmap.org/relation/152072#map=10/48.0331/16.5337
Baden: https://www.openstreetmap.org/relation/111479#map=10/47.9931/16.1856
Wiener Neustadt: https://www.openstreetmap.org/relation/110451#map=10/47.9380/16.2076

You can even see them in your own app if you zoom out until details are not shown, only overview: label selection seems better there, though e.g. "Wiener Neustädter Pforte" shouldn't be shown, it's not a major geographical feature (e.g. only Wikipedia entry is in German).

image

### PR DESCRIPTION
**Pull request by pastk**
Fixes #1995

After the fix the towns in question will be displayed at z9-10
Screenshot_1644007993

### CODE REVIEW

File: data/styles/clear/include/Basemap_label.mapcss [hunk 1]
```diff
@@ -329,14 +329,17 @@ node|z13-[place=city][capital!=2][population>=0]::int_name
...
+node|z9[place=town][population>=20000],
+node|z10[place=town],
 {text: name;text-color: @city_label;}
 node|z8[place=town][population>=40000]::int_name,
-node|z9[place=town][population>=20000]::int_name
+node|z9[place=town][population>=20000]::int_name,
+node|z10[place=town]::int_name
 {text:int_name;text-color: @city_label;}
-node|z10-[place=town][population<20000],
+
```
Comment (biodranik) [inline data/styles/clear/include/Basemap_label.mapcss:339]: ?

**Review by biodranik — APPROVED** [review]

File: data/styles/clear/include/Basemap_label.mapcss (hunk 1 above)
Comment (pastk) [reply data/styles/clear/include/Basemap_label.mapcss:339]: what exactly?
I moved this rule from z10 to z11 to give room in z10 for towns >20k

Comment (biodranik) [reply data/styles/clear/include/Basemap_label.mapcss:339]: an empty line.

**Comment by pastk** [comment]
ATM visibility rules for towns of different populations are like
>=40k  z8-10
>=20k  z9-10
<20k  z10-

File: data/styles/clear/include/Basemap_label.mapcss (hunk 1 above)
Comment (pastk) [reply data/styles/clear/include/Basemap_label.mapcss:339]: Its just to separate sections to improve readability

**Comment by pastk** [comment]
Maybe makes sense to change to make towns >=20k <40k to be visible in 3 zooms at least, e.g. z9-11

**Comment by biodranik** [comment]
> Maybe makes sense to change to make towns >=20k <40k to be visible in 3 zooms at least, e.g. z9-11

Need to test it in countries with different density.

File: data/styles/clear/include/Basemap_label.mapcss [hunk 2]
```diff
@@ -349,10 +352,10 @@ node|z9[place=town][population>=20000]
 node|z9[place=town][population>=20000]::int_name
 {font-size: 9;}
 
-node|z10[place=town][population<20000]
-{font-size: 10;text-color: @city_label;text-halo-opacity: 0.7;}
-node|z10[place=town][population<20000]::int_name
-{font-size: 9;text-color: @city_label;text-halo-opacity: 0.7;}
+node|z10[place=town][population>=0]
```
Comment (vng) [inline data/styles/clear/include/Basemap_label.mapcss:355]: Used simple node|z10[place=town] above.
Is there any difference with [population>=0] ? I'm not sure for features _with and without_ stored population ...

**Review by vng — APPROVED** [review]

**Comment by vng** [comment]
> ATM visibility rules for towns of different populations are like
> 
> ```
> >=40k  z8-10
> >=20k  z9-10
> <20k   z11-
> ```

According to the change set, **all** towns will be displayed in z10.

File: data/styles/clear/include/Basemap_label.mapcss (hunk 2 above)
Comment (pastk) [reply data/styles/clear/include/Basemap_label.mapcss:355]: I used the same approach used for cities in a section above and it seems to be working well.
I'll post more screenshots in a moment.

**Comment by pastk** [comment]
> 
> According to the change set, **all** towns will be displayed in z10.

You're right

**Comment by pastk** [comment]
I have expanded the change to z11.
```
Big towns  z8-11
Mid towns  z9-11
Lil towns   z11-
```
Mid towns are the ones from original issue near Vienna (population >20k <40k). Before the change they were visible at z9 only.

Before
Screenshot_1644056248
After
Screenshot_1644057379

Before
Screenshot_1644056264
After
Screenshot_1644057457

Before
Screenshot_1644056287
After
Screenshot_1644057507

**Comment by pastk** [comment]
Before
Screenshot_1644056367
After
Screenshot_1644058699

Before
Screenshot_1644056382
After
Screenshot_1644058729

Before:
Screenshot_1644056191
After:
Screenshot_1644057294

**Comment by vng** [comment]
To clarify, Little towns from z10.

**Comment by pastk** [comment]
> To clarify, Little towns from z10.

You're right again, fixed in my comment above, thanks!

**Comment by pastk** [comment]
To further clarify
```
           before      after
Big towns  z8-9        z8-11
Mid towns  z9          z9-11
Lil towns  z10-        z10-
```

**Review by vng — APPROVED** [review]

**Pull request merged**